### ISSUE TITLE
Ganeti 2.0 doesn't check host name with new version of hostname

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#128]
Originally reported of Google Code with ID 72.

```
<b>What steps will reproduce the problem?</b>
1. Ganeti 2.0, hostname version 2.97 and more
2. Try to add a new node with gnt-node add nodeX.host.gnt 
3. Fail with check hostname error "hostname mismatch, got nodeX"

The reason is http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=420053 which
was corrected in 2.97 release. So now, hostname checks if you have a fqdn
in /etc/hostname (you should not have it, even if ganeti documentation say
you should) and truncate it to  return the short name.

You should use hostname -f to get the fqdn. 

Patch for this is attached, works at least for me :)

Cheers,

Maxence
```
Originally added on 2009-10-06 16:51:17 +0000 UTC.

Attachments: 
[PATCH_invalid_hostname_call.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-72/comment-0/PATCH_invalid_hostname_call.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2009-10-23 12:44:44 +0000 UTC.

Added to Milestone: Release2.1

**Comment by Ganeti-Issues-Migrator** [comment]
```
This was fixed in the 2.1 release.

thanks,
iustin
```
Originally added on 2010-03-06 18:43:35 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
